transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java

```diff
@@ -829,4 +810,25 @@ Entry recycleAndGetNext() {
             return next;
         }
     }
+
+    private static final class ChannelWritabilityChangedTask implements Runnable {
+        private final Channel channel;
+        private boolean writable = true;
+
+        ChannelWritabilityChangedTask(Channel channel) {
+            this.channel = channel;
+        }
+
+        @Override
+        public void run() {
+            if (channel.isActive()) {
+                boolean newWritable = channel.isWritable();
+
+                if (writable != newWritable) {
```

Comment:
Staring at this I can't tell if it is racy or not. Since every change will add the runnable to the event loop, I _guess_ it would eventually still fire, even if the writability changed in "random" ways between the scheduling and the execution. It seems that it can also be "reordered" in that _later_ changed events are squelched in favor of _earlier_ notifications, which is a bit counter-intuitive because the _earlier_ notification is the one that was invalidated by the later state change. However, all that seems "okay" and within the permitted API and I can't think of any ways it would cause problems.

So, I _think_ it is fine and I think it would work pretty well (so I'm not encouraging changing it), but I didn't know if me pointing out some oddities here would make something obvious to someone else.